docs/guide/api-environment-plugins.md

```diff
@@ -31,11 +31,19 @@ A plugin could use the `environment` instance to change how a module is processe
 
 ## Registering New Environments Using Hooks
 
-Plugins can add new environments in the `config` hook (for example to have a separate module graph for [RSC](https://react.dev/blog/2023/03/22/react-labs-what-we-have-been-working-on-march-2023#react-server-components)):
+Plugins can add new environments in the `config` hook. For example, [RSC support](/guide/features#react-server-components) uses an additional environment to have a separate module graph for `react-server` condition by:
 
 ```ts
   config(config: UserConfig) {
-    config.environments.rsc ??= {}
+    return {
+      environments: {
+        rsc: {
+          resolve: {
+            conditions: ['react-server', ...defaultServerConditions],
+          },
+        },
+      },
+    }
```

Comment:
Setting the conditions this way can be a little tricky I think depending on whether the `config` already have an existing value. If it does and have conditions explicitly not included in `defaultServerConditions`, then this config will add the condition back due to config merging.

The way the svelte plugin handles it is with something [like this](https://github.com/sveltejs/vite-plugin-svelte/blob/3a6344f11c4d6f7fba8da85776ac135bea82e1a0/packages/vite-plugin-svelte/src/plugins/configure.js#L87-L89) and it is kinda complicated. Maybe it would be easier with a different example here?

Reply:
Thanks for the reference. Right, it looks like what svelte does is a correct way to add conditions. I guess that needs a proper documentation separately.

FWIW, this is the actual logic `@vitejs/plugin-rsc` does and it's mostly safe for rsc environment use cases, but let me think if I can find simpler config example.